test/unit_test/passes/onnx/test_optimum_conversion.py

```diff
@@ -12,14 +12,45 @@
 from olive.passes.onnx.optimum_conversion import OptimumConversion
 
 
-@pytest.mark.parametrize("input_model", [get_optimum_model_by_hf_config(), get_optimum_model_by_model_path()])
-def test_optimum_conversion_pass(input_model, tmp_path):
+@pytest.mark.parametrize(
+    "input_model,extra_args",
+    [(get_optimum_model_by_hf_config(), {"atol": 0.1}), (get_optimum_model_by_model_path(), {"atol": None})],
+)
+def test_optimum_conversion_pass(input_model, extra_args, tmp_path):
     # setup
-    p = create_pass_from_dict(OptimumConversion, {}, disable_search=True)
+    p = create_pass_from_dict(OptimumConversion, {"extra_args": extra_args}, disable_search=True)
     output_folder = tmp_path
 
     # execute
     onnx_model = p.run(input_model, None, output_folder)
 
     # assert
     assert Path(onnx_model.model_path).exists()
+
+
+@pytest.mark.parametrize(
+    "config,is_valid",
+    [
+        ({"fp16": True}, False),
+        ({"fp16": True, "device": "cpu"}, False),
+        ({"fp16": True, "device": "cuda"}, True),
+        ({"fp16": False, "device": "cuda"}, True),
+    ],
+)
+def test_optimum_configs(config, is_valid, tmp_path):
+    input_model = get_optimum_model_by_hf_config()
+    # setup
+    p = create_pass_from_dict(OptimumConversion, config, disable_search=True)
+    output_folder = tmp_path
+
+    if not is_valid:
+        assert p.validate_search_point(config, None) is False
+        with pytest.raises(
```

Comment:
nit: is this part needed? 

I think matching a specific error message makes the test breakable if optimum changes the error message. Or is this the purpose? So that we can catch any incompatibility with new optimum changes?

Reply:
I also try to remove the match part in my first version of implementation. But as you said, once they change the code and have different supports on fp16, then we can catch that.